<img src="Logo.png" align="right" width="150" />

##
***
## (950457) Medidas Electrónicas I
#####

Profesor: Ing. Emiliano Marinsek	<<emarinsek@frba.utn.edu.ar>> <br>
Ayudante: Ing. Juan Manuel Perdomo	<<jmperdomo@frba.utn.edu.ar>>

***
####
### <u> Ejercicios de incertidumbre</u>
##### *Introducción al cálculo de incertidumbre*
####

Por David Moharos

***

<img src="Consigna.png" style="margin-left:auto; margin-right:auto"/>

In [214]:
#%% Librerías y módulos de Python a utilizar

import numpy as np
from IPython.display import display, Markdown, HTML
import pandas as pd

In [175]:
#%% Métodos auxiliares

#-- Ejercicio 1 --
def calculo_em (R1, R2):
    return ( R2 / (R2 - R1) )

#-- Ejercicio 2 --
def calculo_deltaV (V1, V2):
    return ( V1 - V2 )

#-- Ejercicio 3 --
def calculo_Vout (Voff, a_s, g, theta):
    return ( Voff + a_s / g * g * np.sin(theta) )

#-- Ejercicio 4 --
def calculo_R1 (V1, I1):
    return ( V1 / I1 )
    
#-- Ejercicio 5--
def calculo_V0 (H1, H0, KH, S, RG):
    AV = 1 + 50 * 10**3 / RG
    return ( (H1 - H0) * KH * S * AV )

#-- Métodos comunes a todos los ejercicios --
def valor_mas_probable (mediciones):
    return np.mean(mediciones, dtype=np.float64)

def incertidumbre_tipo_A (mediciones):
    s = np.std(mediciones, ddof=1, )     # ddof=1 -> grados de libertad v = 1/n-1
    return s/np.sqrt(len(mediciones))

def incertidumbre_tipo_B (error_relativo, error_cuentas, cuentas_indicadas, mediciones):
    return (error_relativo/100 + error_cuentas/cuentas_indicadas) * np.mean(mediciones) / np.sqrt(3)

def incertidumbre_tipo_B (error_relativo, error_cuentas, cuentas_indicadas, valor_mas_probable):
    return (error_relativo/100 + error_cuentas/cuentas_indicadas) * valor_mas_probable / np.sqrt(3)

def incertidumbre_combinada (uA, uB):
    return np.sqrt(uA**2 + uB**2)

<img src="Ejercicio_1.png" style="margin-left:auto; margin-right:auto"/>

***
#####

#### <u>Cálculo de valor más probable</u>

#####

Se obtienen los valores más probables de $R_1$ y $R_2$ a través de la siguiente expresión:

#####

\begin{equation*}
\overline{R_i} = \frac{1}{n} \cdot \sum\limits_{k=1}^{n} R_{i,k}
\end{equation*}

#####

Siendo:

> n: cantidad de mediciones

> $R_{i_{k}}$: i-ésimos valores de resistencias, k-ésimos valores de las n mediciones.

#####

Reemplazando los valores de la tabla, obtenemos $\overline{R_1}$ y $\overline{R_2}$:

#####

In [158]:
R1 = [2.01, 2.04, 2.03, 2.02, 2.04]
R2 = [0.53, 0.54, 0.52, 0.51, 0.53]

ov_R1 = valor_mas_probable(R1)
ov_R2 = valor_mas_probable(R2) 

display(Markdown("##\n" + f"$$\\overline{{R_1}} = {ov_R1} \, [\Omega]$$\n"))
display(Markdown("##\n" + f"$$\\overline{{R_2}} = {ov_R2} \, [\Omega]$$\n"))

##
$$\overline{R_1} = 2.028 \, [\Omega]$$


##
$$\overline{R_2} = 0.526 \, [\Omega]$$


#####

Se obtiene el valor más probable de $e_m$ a través de la siguiente expresión:

#####

\begin{equation*}
\overline{e_m} = \frac{ \overline{ R_2 } }{ \overline{ R_2 } - \overline{ R_1 } }
\end{equation*}

#####

In [159]:
ov_em = calculo_em (ov_R1, ov_R2)

display(Markdown("##\n" + f"$$\\overline{{e_m}} = {ov_em}$$\n"))

##
$$\overline{e_m} = -0.35019973368841545$$


#####

#### <u>Cálculo de incertidumbres</u>

#####

<u>Incertidumbre tipo A:</u>

#####

Se obtienen las incertidumbres tipo A a través de la siguiente expresión:

#####

\begin{equation*}
u_A(R_i) = \frac{ s(R_i) }{ \sqrt{n} } = \frac{ \sqrt{ \frac{ 1 }{ n-1 } \sum \limits_{k=1}^{n} \left( R_{i,k} - \overline{R_i} \right)^{2} } }{ \sqrt{n} }
\end{equation*}

#####

In [153]:
uA_R1 = incertidumbre_tipo_A (R1)
uA_R2 = incertidumbre_tipo_A (R2)

display(Markdown("##\n" + f"$$u_A(R_1) = {uA_R1} \, [\Omega]$$\n"))
display(Markdown("##\n" + f"$$u_A(R_2) = {uA_R2} \, [\Omega]$$\n"))

##
$$u_A(R_1) = 0.005830951894845336 \, [\Omega]$$


##
$$u_A(R_2) = 0.00509901951359279 \, [\Omega]$$


#####

<u>Incertidumbre tipo B:</u>

#####

Se obtienen las incertidumbres tipo B a través de la siguiente expresión:

#####

\begin{equation*}
u_B(R_i) = \frac{ \Delta R_i }{ \sqrt{3} } = \frac{ \left( \frac{er\%}{100} + \frac{n}{ci} \right) \cdot \overline{R_i} }{ \sqrt{3} } 
\end{equation*}

#####


Siendo:

> $er%$: Error relativo.

> $n$: Error de cuentas.

> $ci$: Cuentas indicadas.

#####

Además:

> $ci = \frac{Ind}{Res} \frac{ [u] }{ \left[ \frac{u}{cuenta} \right] }= [cuentas]$

> $Ind$: Indicación del instrumento con su respectiva $u$ unidad.

> $Res$: Resolución del instrumento con su respectiva $u$ unidad por cuenta, dada por el rango seleccionado y la cantidad de dígitos que el mismo puede representar.

#####


In [154]:
er = 0.5
n = 2

ind_R1 = ov_R1
res_R1 = 0.01
ci_R1 = ind_R1 / res_R1

ind_R2 = ov_R2
res_R2 = 0.01
ci_R2 = ind_R2 / res_R2

#def incertidumbre_tipo_B (error_relativo, error_cuentas, cuentas_indicadas, valor_mas_probable):
uB_R1 = incertidumbre_tipo_B (er, n, ci_R1, ov_R1)
uB_R2 = incertidumbre_tipo_B (er, n, ci_R2, ov_R2)

display(Markdown("##\n" + f"$$u_B(R_1) = {uB_R1} \, [\Omega]$$\n"))
display(Markdown("##\n" + f"$$u_B(R_2) = {uB_R2} \, [\Omega]$$\n"))

##
$$u_B(R_1) = 0.017401337113375322 \, [\Omega]$$


##
$$u_B(R_2) = 0.013065436591761232 \, [\Omega]$$


#####

<u>Incertidumbre de $R_i$:</u>

#####

Se obtienen las incertidumbres de $R_i$ a través de la siguiente expresión:

#####

\begin{equation*}
u(R_i) = \sqrt{ u_A^2(R_i) + u_B^2(R_i) }
\end{equation*}

#####

In [155]:
#def incertidumbre_combinada (uA, uB):
uC_R1 = incertidumbre_combinada (uA_R1, uB_R1)
uC_R2 = incertidumbre_combinada (uA_R2, uB_R2)

display(Markdown("##\n" + f"$$u(R_1) = {uC_R1} \, [\Omega]$$\n"))
display(Markdown("##\n" + f"$$u(R_2) = {uC_R2} \, [\Omega]$$\n"))

##
$$u(R_1) = 0.018352289593762786 \, [\Omega]$$


##
$$u(R_2) = 0.01402517854907143 \, [\Omega]$$


#####

<u> Incertidumbre combinada </u>

#####

Se obtienen la incertidumbre combinada de $e_m$ a través de la siguiente expresión:

#####

\begin{equation*} 
u_C(e_m) = \sqrt{\left[\frac{de_m}{dR_1}\right]^2 \cdot u^2(R_1) + \left[\frac{de_m}{dR_2}\right]^2 \cdot u^2(R_2) }
\end{equation*}

#####

\begin{equation*} 
u_C(e_m) = \sqrt{\left[ \frac{ \overline{R_2} }{ ( \overline{R_2} - \overline{R_1} )^2 } \right]^2 \cdot u^2(R_1) + \left[ \frac{ -\overline{R_1} }{ ( \overline{R_2} - \overline{R_1} )^2 } \right]^2 \cdot u^2(R_2) }
\end{equation*}

#####

In [156]:
dem_dR1 = ov_R2 / (ov_R2 - ov_R1)**2
dem_dR2 = -ov_R1 / (ov_R2 - ov_R1)**2

uC_em = np.sqrt( (dem_dR1)**2 * uC_R1**2 + (dem_dR2)**2 * uC_R2**2)

display(Markdown("##\n" + f"$$u_C(e_m) = {uC_em}$$\n"))

##
$$u_C(e_m) = 0.013314047982818046$$


#####

<u>Incertidumbre expandida:</u>

#####

\begin{equation*}
U(e_m) = u_C(e_m) \cdot k_{95\%}
\end{equation*}

#####

In [157]:
k_95 = 2
U_em = uC_em * k_95

display(Markdown("##\n" + f"$$U(e_m) = {U_em}$$\n" + "##"))

##
$$U(e_m) = 0.026628095965636093$$
##

#####

<u>Resultado de la medición:</u>

#####

\begin{equation*}
e_m = \overline{e_m} \pm U(e_m) 
\end{equation*}

#####

\begin{equation*}
e_m = \left( -0.35019973368841545 \pm 0.026628095965636093 \right) \qquad k = 2
\end{equation*}

#####

\begin{equation*}
\boxed{ e_m = \left( -0.350 \pm 0.027 \right) \qquad k = 2 }
\end{equation*}

#####

***

<img src="Ejercicio_2.png" style="margin-left:auto; margin-right:auto"/>

***
#####

#### <u>Cálculo de valor más probable</u>

#####

Se obtienen los valores más probables de $V_1$ y $V_2$ a través de la siguiente expresión:

#####

\begin{equation*}
\overline{V_i} = \frac{1}{n} \cdot \sum\limits_{k=1}^{n} V_{i,k}
\end{equation*}

#####

Siendo:

> n: cantidad de mediciones

> $V_{i_{k}}$: i-ésimos valores de tensiones, k-ésimos valores de las n mediciones.

#####

Reemplazando los valores de la tabla, obtenemos $\overline{V_1}$ y $\overline{V_2}$:

#####

In [162]:
V1 = [2.404, 2.502, 2.603, 2.104, 2.204]
V2 = [0.54, 0.58, 0.51, 0.50, 0.56]

ov_V1 = valor_mas_probable(V1)
ov_V2 = valor_mas_probable(V2)

display(Markdown("##\n" + f"$$\\overline{{V_1}} = {ov_V1} \, [V]$$\n"))
display(Markdown("##\n" + f"$$\\overline{{V_2}} = {ov_V2} \, [V]$$\n"))

##
$$\overline{V_1} = 2.3634 \, [V]$$


##
$$\overline{V_2} = 0.538 \, [V]$$


#####

Se obtiene el valor más probable de $\Delta V$ a través de la siguiente expresión:

#####

\begin{equation*}
\overline{\Delta V} = \overline{V_1} - \overline{V_2}
\end{equation*}

#####

In [163]:
ov_DeltaV = calculo_deltaV (ov_V1, ov_V2)

display(Markdown("##\n" + f"$$\\overline{{\\Delta V}} = {ov_DeltaV} \, [V]$$\n"))

##
$$\overline{\Delta V} = 1.8254 \, [V]$$


#####

#### <u>Cálculo de incertidumbres</u>

#####

<u>Incertidumbre tipo A:</u>

#####

Se obtienen las incertidumbres tipo A a través de la siguiente expresión:

#####

\begin{equation*}
u_A(V_i) = \frac{ s(V_i) }{ \sqrt{n} } = \frac{ \sqrt{ \frac{ 1 }{ n-1 } \sum \limits_{k=1}^{n} \left( V_{i,k} - \overline{V_i} \right)^{2} } }{ \sqrt{n} }
\end{equation*}

#####

In [220]:
uA_V1 = incertidumbre_tipo_A (V1)
uA_V2 = incertidumbre_tipo_A (V2)

display(Markdown("##\n" + f"$$u_A(V_1) = {uA_V1} \, [V]$$\n"))
display(Markdown("##\n" + f"$$u_A(V_2) = {uA_V2} \, [V]$$\n"))

##
$$u_A(V_1) = 0.0009797958971132583 \, [V]$$


##
$$u_A(V_2) = 0.014966629547095763 \, [V]$$


#####

<u>Incertidumbre tipo B:</u>

#####

Se obtienen las incertidumbres tipo B a través de la siguiente expresión:

#####

\begin{equation*}
u_B(V_i) = \frac{ \Delta V_i }{ \sqrt{3} } = \frac{ \left( \frac{er\%}{100} + \frac{n}{ci} \right) \cdot \overline{V_i} }{ \sqrt{3} } 
\end{equation*}

#####


Siendo:

> $er%$: Error relativo.

> $n$: Error de cuentas.

> $ci$: Cuentas indicadas.

#####

Además:

> $ci = \frac{Ind}{Res} \frac{ [u] }{ \left[ \frac{u}{cuenta} \right] }= [cuentas]$

> $Ind$: Indicación del instrumento con su respectiva $u$ unidad.

> $Res$: Resolución del instrumento con su respectiva $u$ unidad por cuenta, dada por el rango seleccionado y la cantidad de dígitos que el mismo puede representar.

#####

In [218]:
er_V1 = 1
n_V1 = 2

ind_V1 = ov_V1
res_V1 = 0.001
ci_V1 = ind_V1 / res_V1

er_V2 = 0.5 
n_V2 = 4

ind_V2 = ov_V2
res_V2 = 0.01
ci_V2 = ind_V2 / res_V2

#def incertidumbre_tipo_B (error_relativo, error_cuentas, cuentas_indicadas, valor_mas_probable):
uB_V1 = incertidumbre_tipo_B (er_V1, n_V1, ci_V1, ov_V1)
uB_V2 = incertidumbre_tipo_B (er_V2, n_V2, ci_V2, ov_V2)

display(Markdown("##\n" + f"$$u_B(V_1) = {uB_V1} \, [V]$$\n"))
display(Markdown("##\n" + f"$$u_B(V_2) = {uB_V2} \, [V]$$\n"))

##
$$u_B(V_1) = 0.012831032382470246 \, [V]$$


##
$$u_B(V_2) = 0.024647082991705125 \, [V]$$


#####

<u>Incertidumbre de $V_i$:</u>

#####

Se obtienen las incertidumbres de $V_i$ a través de la siguiente expresión:

#####

\begin{equation*}
u(V_i) = \sqrt{ u_A^2(V_i) + u_B^2(V_i) }
\end{equation*}

#####

In [26]:
#def incertidumbre_combinada (uA, uB):
uC_V1 = incertidumbre_combinada (uA_V1, uB_V1)
uC_V2 = incertidumbre_combinada (uA_V2, uB_V2)

display(Markdown("##\n" + f"$$u(V_1) = {uC_V1} \, [V]$$\n"))
display(Markdown("##\n" + f"$$u(V_2) = {uC_V2} \, [V]$$\n"))

##
$$u(V_1) = 0.09363329528182446 \, [V]$$


##
$$u(V_2) = 0.028835372374914806 \, [V]$$


#####

<u> Incertidumbre combinada </u>

#####

Se obtiene la incertidumbre combinada de $\Delta V$ a través de la siguiente expresión:

#####

\begin{equation*} 
u_C(\Delta V) = \sqrt{\left[\frac{d\Delta V}{dV_1}\right]^2 \cdot u^2(V_1) + \left[\frac{d\Delta V}{dV_2}\right]^2 \cdot u^2(V_2) }
\end{equation*}

#####

\begin{equation*} 
u_C(\Delta V) = \sqrt{\left[ 1 \right]^2 \cdot u^2(V_1) + \left[ -1 \right]^2 \cdot u^2(V_2) }
\end{equation*}

#####

In [27]:
dDeltaV_dV1 = 1
dDeltaV_dV2 = -1

uC_DeltaV = np.sqrt( (dDeltaV_dV1)**2 * uC_V1**2 + (dDeltaV_dV2)**2 * uC_V2**2)

display(Markdown("##\n" + f"$$u_C(\\Delta V) = {uC_DeltaV} \, [V]$$\n"))

##
$$u_C(\Delta V) = 0.09797281605288954 \, [V]$$


#####

<u>Incertidumbre expandida:</u>

#####

\begin{equation*}
U(\Delta V) = u_C(\Delta V) \cdot k_{95\%}
\end{equation*}

#####

In [34]:
k_95 = 2
U_DeltaV = uC_DeltaV * k_95

display(Markdown("##\n" + f"$$U(\\Delta V) = {U_DeltaV} \, [V]$$\n" + "##"))

##
$$U(\Delta V) = 0.1959456321057791 \, [V]$$
##

#####

<u>Resultado de la medición:</u>

#####

<u>Aproximación general:</u>

#####

\begin{equation*}
\Delta V = \overline{\Delta V} \pm U(\Delta V) 
\end{equation*}

#####

\begin{equation*}
\Delta V = \left( 1.8254 \pm 0.1959456321057791 \right) [V] \qquad k = 2
\end{equation*}

#####

\begin{equation*}
\boxed{ \Delta V = \left( 1.82 \pm 0.20 \right) [V] \qquad k = 2 }
\end{equation*}

#####

***

#####

Considerando los diferentes aportantes de distribuciones...

#####

<u>Grados de libertad</u>

#####

\begin{equation*}
\nu_{eff} = \frac{ u_C^4(y) }{ \sum \limits_{k=1}^{n} \frac{ u_i^4(y) }{ \nu_i } }
\end{equation*}

#####

\begin{equation*}
u_i(y) = \frac{ dy }{ dx_i } \cdot u(x_i)
\end{equation*}

#####

\begin{equation*}
\nu_i = n_i - 1
\end{equation*}

#####

\begin{equation*}
\nu_{eff} = \frac{ u_C^4( \Delta V ) }
{ \frac{ \left[ \frac{ d \Delta V }{ V_1 } \cdot u_A(V_1) \right]^4 }{ \nu_{AV_{1}} } +
\frac{ \left[ \frac{ d \Delta V }{ V_1 } \cdot u_B(V_1) \right]^4 }{ \nu_{BV_{1}} } +
\frac{ \left[ \frac{ d \Delta V }{ V_2 } \cdot u_A(V_2) \right]^4 }{ \nu_{AV_{2}} } +
\frac{ \left[ \frac{ d \Delta V }{ V_2 } \cdot u_B(V_2) \right]^4 }{ \nu_{BV_{2}} } } 
\end{equation*}

#####

\begin{equation*}
\nu_{eff} = \frac{ u_C^4( \Delta V ) }
{ \frac{ \left[ 1 \cdot u_A(V_1) \right]^4 }{ \nu_{AV_{1}} } +
\frac{ \left[ 1 \cdot u_B(V_1) \right]^4 }{ \nu_{BV_{1}} } +
\frac{ \left[ (-1) \cdot u_A(V_2) \right]^4 }{ \nu_{AV_{2}} } +
\frac{ \left[ (-1) \cdot u_B(V_2) \right]^4 }{ \nu_{BV_{2}} } } 
\end{equation*}

#####

\begin{equation*}
\nu_{eff} = \frac{ u_C^4( \Delta V ) }
{ \frac{ \left[ 1 \cdot 0.0009797958971132583 \right]^4 }{ 5 - 1 } +
\frac{ \left[ 1 \cdot 0.012831032382470246 \right]^4 }{ \infty } +
\frac{ \left[ (-1) \cdot 0.014966629547095763 \right]^4 }{ 5 - 1 } +
\frac{ \left[ (-1) \cdot 0.024647082991705125 \right]^4 }{ \infty } }
\end{equation*}

#####

In [239]:
v_AV1 = len(V1) - 1
v_BV1 = np.inf
v_AV2 = len(V2) - 1 
v_BV2 = np.inf

a = (dDeltaV_dV1 * uA_V1)**4
b = (dDeltaV_dV2 * uB_V1)**4
c = (dDeltaV_dV1 * uA_V2)**4
d = (dDeltaV_dV2 * uB_V2)**4

v_eff = uC_DeltaV**4 / ( (a/v_AV1) + (b/v_BV1) + (c/v_AV2) + (d/v_BV2) )

display(Markdown("##\n" + f"$$\\nu_{{eff}} = {v_eff}$$\n" + "##"))

##
$$\nu_{eff} = 7344.772407892337$$
##

<img src="Ejercicio_3.png" style="margin-left:auto; margin-right:auto"/>

***

#####

#### <u>Cálculo de valor más probable</u>

#####

Se obtiene el valor más probable de $V_{OUT}$ a través de la siguiente expresión:

#####

\begin{equation*}
\overline{V_{OUT}} = \overline{V_{OFF}} + a_s \cdot g \cdot sen\left(\overline{\theta}\right) 
\end{equation*}

#####

Reemplazando los valores de la tabla, obtenemos $\overline{V_{OUT}}$:

#####

In [165]:
ov_Voff = 20.03            # [V]
a_s = 800 * 10**-3         # [V/g]
g = 9.80665                # [m/s^2]
ov_theta = 30.028          # [rad]

#def calculo_Vout (Voff, a_s, g, theta)
ov_Vout = calculo_Vout (ov_Voff, a_s, g, ov_theta)

display(Markdown("##\n" + f"$$\\overline{{V_{{OUT}}}} = {ov_Vout} \, [V]$$\n"))

##
$$\overline{V_{OUT}} = 19.243339308211706 \, [V]$$


#####

#### <u>Cálculo de incertidumbres</u>

#####

<u>Incertidumbre tipo A:</u>

#####

La incertidumbre tipo A viene dada directamente como dato a través del desvío estándar de la media:

#####

In [63]:
uA_Voff = 0.01581            # [V]
uA_theta = 0.001304          # [rad]

display(Markdown("##\n" + f"$$u_A(V_{{OFF}}) = {uA_Voff} \, [V]$$\n"))
display(Markdown("##\n" + f"$$u_A(\\theta) = {uA_theta} \, [rad]$$\n"))

##
$$u_A(V_{OFF}) = 0.01581 \, [V]$$


##
$$u_A(\theta) = 0.001304 \, [rad]$$


#####

<u>Incertidumbre tipo B:</u>

#####

Se obtienen las incertidumbres tipo B a través de la siguiente expresión:

#####

\begin{equation*}
u_B(X) = \frac{ \Delta X }{ \sqrt{3} } = \frac{ \left( \frac{er\%}{100} + \frac{n}{ci} \right) \cdot \overline{X} }{ \sqrt{3} } 
\end{equation*}

#####


Siendo:

> $er%$: Error relativo.

> $n$: Error de cuentas.

> $ci$: Cuentas indicadas.

#####

Además:

> $ci = \frac{Ind}{Res} \frac{ [u] }{ \left[ \frac{u}{cuenta} \right] }= [cuentas]$

> $Ind$: Indicación del instrumento con su respectiva $u$ unidad.

> $Res$: Resolución del instrumento con su respectiva $u$ unidad por cuenta, dada por el rango seleccionado y la cantidad de dígitos que el mismo puede representar.

#####

In [73]:
er_Voff = 0.5
n_Voff = 2

ind_Voff = ov_Voff
res_Voff = 0.01
ci_Voff = ind_Voff / res_Voff

er_theta = 0.1
n_theta = 0.5

ind_theta = ov_theta
res_theta = 0.001
ci_theta = ind_theta / res_theta

#def incertidumbre_tipo_B (error_relativo, error_cuentas, cuentas_indicadas, valor_mas_probable):
uB_Voff = incertidumbre_tipo_B (er_Voff, n_Voff, ci_Voff, ov_Voff)
uB_theta = incertidumbre_tipo_B (er_theta, n_theta, ci_theta, ov_theta)

display(Markdown("##\n" + f"$$u_B(V_{{OFF}}) = {uB_Voff} \, [V]$$\n"))
display(Markdown("##\n" + f"$$u_B(\\theta) = {uB_theta} \, [rad]$$\n"))

##
$$u_B(V_{OFF}) = 0.06936863484313355 \, [V]$$


##
$$u_B(\theta) = 0.017625349017820896 \, [rad]$$


#####

<u>Incertidumbre de $V_{OFF}$ y $\theta$:</u>

#####

Se obtienen las incertidumbres de $V_{OFF}$ y $\theta$ a través de la siguiente expresión:

#####

\begin{equation*}
u(X) = \sqrt{ u_A^2(X) + u_B^2(X) }
\end{equation*}

#####

In [70]:
#def incertidumbre_combinada (uA, uB):
u_Voff = incertidumbre_combinada (uA_Voff, uB_Voff)
u_theta = incertidumbre_combinada (uA_theta, uB_theta)

display(Markdown("##\n" + f"$$u(V_{{OFF}}) = {u_Voff} \, [V]$$\n"))
display(Markdown("##\n" + f"$$u(\\theta) = {u_theta} \, [rad]$$\n"))

##
$$u(V_{OFF}) = 0.07114747781896419 \, [V]$$


##
$$u(\theta) = 0.017673520984795306 \, [rad]$$


#####

<u> Incertidumbre combinada </u>

#####

Se obtiene la incertidumbre combinada de $V_{OUT}$ a través de la siguiente expresión:

#####

\begin{equation*} 
u_C(V_{OUT}) = \sqrt{\left[\frac{dV_{OUT}}{dV_{OFF}}\right]^2 \cdot u^2(V_{OFF}) + \left[\frac{dV_{OUT}}{d\theta}\right]^2 \cdot u^2(\theta) }
\end{equation*}

#####

\begin{equation*} 
u_C(V_{OUT}) = \sqrt{\left[ 1 \right]^2 \cdot u^2(V_{OFF}) + \left[ a_s \cdot g \cdot cos(\theta) \right]^2 \cdot u^2(\theta) }
\end{equation*}

#####

In [72]:
dVout_dVoff = 1
dVout_dtheta = a_s / g * g * np.cos(ov_theta)

uC_Vout = np.sqrt( (dVout_dVoff)**2 * u_Voff**2 + (dVout_dtheta)**2 * u_theta**2)

display(Markdown("##\n" + f"$$u_C(V_{{OUT}}) = {uC_Vout} \, [V]$$\n"))

##
$$u_C(V_{OUT}) = 0.07119392210561729 \, [V]$$


#####

<u>Incertidumbre expandida:</u>

#####

\begin{equation*}
U(V_{OUT}) = u_C(V_{OUT}) \cdot k_{95\%}
\end{equation*}

#####

In [75]:
k_95 = 2
U_Vout = uC_Vout * k_95

display(Markdown("##\n" + f"$$U(V_{{OUT}}) = {U_Vout} \, [V]$$\n" + "##"))

##
$$U(V_{OUT}) = 0.14238784421123457 \, [V]$$
##

#####

<u>Resultado de la medición:</u>

#####

\begin{equation*}
V_{OUT} = \overline{V_{OUT}} \pm U(V_{OUT}) 
\end{equation*}

#####

\begin{equation*}
V_{OUT} = \left( 19.243339308211706 \pm 0.14238784421123457 \right) [V] \qquad k = 2
\end{equation*}

#####

\begin{equation*}
\boxed{ V_{OUT} = \left( 19,24 \pm 0.14 \right) [V] \qquad k = 2 }
\end{equation*}

#####

***

<img src="Ejercicio_4.png" style="margin-left:auto; margin-right:auto"/>

***
#####

#### <u>Cálculo de valor más probable</u>

#####

Se obtienen los valores más probables de $V_1$ y $I_1$ a través de la siguiente expresión:

#####

\begin{equation*}
\overline{X_i} = \frac{1}{n} \cdot \sum\limits_{k=1}^{n} X_{i,k}
\end{equation*}

#####

Siendo:

> n: cantidad de mediciones

> $X_{i_{k}}$: i-ésimos valores de tensiones y/o corrientes, k-ésimos valores de las n mediciones.

#####

Reemplazando los valores de la tabla, obtenemos $\overline{V_1}$ e $\overline{I_1}$:

#####

In [167]:
V1 = [2.021, 2.023, 2.026, 2.021, 2.021]
I1 = [0.541, 0.542, 0.544, 0.543, 0.538]

ov_V1 = valor_mas_probable(V1)
ov_I1 = valor_mas_probable(I1) 

display(Markdown("##\n" + f"$$\\overline{{V_1}} = {ov_V1} \, [V]$$\n"))
display(Markdown("##\n" + f"$$\\overline{{I_1}} = {ov_I1} \, [A]$$\n"))

##
$$\overline{V_1} = 2.0224 \, [V]$$


##
$$\overline{I_1} = 0.5416000000000001 \, [A]$$


#####

Se obtiene el valor más probable de $R_1$ a través de la siguiente expresión:

#####

\begin{equation*}
\overline{R_1} = \frac{ \overline{V_1} }{ \overline{I_1} }
\end{equation*}

#####

In [171]:
ov_R1 = calculo_R1 (ov_V1, ov_I1)

display(Markdown("##\n" + f"$$\\overline{{R_1}} = {ov_R1} \, [\Omega]$$\n"))

##
$$\overline{R_1} = 3.734121122599704 \, [\Omega]$$


#####

#### <u>Cálculo de incertidumbres</u>

#####

<u>Incertidumbre tipo A:</u>

#####

Se obtienen las incertidumbres tipo A a través de la siguiente expresión:

#####

\begin{equation*}
u_A(X_i) = \frac{ s(X_i) }{ \sqrt{n} } = \frac{ \sqrt{ \frac{ 1 }{ n-1 } \sum \limits_{k=1}^{n} \left( X_{i,k} - \overline{X_i} \right)^{2} } }{ \sqrt{n} }
\end{equation*}

#####

In [82]:
uA_V1 = incertidumbre_tipo_A (V1)
uA_I1 = incertidumbre_tipo_A (I1)

display(Markdown("##\n" + f"$$u_A(V_1) = {uA_V1} \, [V]$$\n"))
display(Markdown("##\n" + f"$$u_A(I_1) = {uA_I1} \, [A]$$\n"))

##
$$u_A(V_1) = 0.0009797958971132583 \, [V]$$


##
$$u_A(I_1) = 0.001029563014098701 \, [A]$$


#####

<u>Incertidumbre tipo B:</u>

#####

Se obtienen las incertidumbres tipo B a través de la siguiente expresión:

#####

\begin{equation*}
u_B(X_i) = \frac{ \Delta X_i }{ \sqrt{3} } = \frac{ \left( \frac{er\%}{100} + \frac{n}{ci} \right) \cdot \overline{X_i} }{ \sqrt{3} } 
\end{equation*}

#####


Siendo:

> $er%$: Error relativo.

> $n$: Error de cuentas.

> $ci$: Cuentas indicadas.

#####

Además:

> $ci = \frac{Ind}{Res} \frac{ [u] }{ \left[ \frac{u}{cuenta} \right] }= [cuentas]$

> $Ind$: Indicación del instrumento con su respectiva $u$ unidad.

> $Res$: Resolución del instrumento con su respectiva $u$ unidad por cuenta, dada por el rango seleccionado y la cantidad de dígitos que el mismo puede representar.

#####

In [83]:
er_V1 = 0.5
n_V1 = 1

ind_V1 = ov_V1
res_V1 = 0.001
ci_V1 = ind_V1 / res_V1

er_I1 = 1 
n_I1 = 3

ind_I1 = ov_I1
res_I1 = 0.001
ci_I1 = ind_I1 / res_I1

#def incertidumbre_tipo_B (error_relativo, error_cuentas, cuentas_indicadas, valor_mas_probable):
uB_V1 = incertidumbre_tipo_B (er_V1, n_V1, ci_V1, ov_V1)
uB_I1 = incertidumbre_tipo_B (er_I1, n_I1, ci_I1, ov_I1)

display(Markdown("##\n" + f"$$u_B(V_1) = {uB_V1} \, [V]$$\n"))
display(Markdown("##\n" + f"$$u_B(I_1) = {uB_I1} \, [A]$$\n"))

##
$$u_B(V_1) = 0.006415516191235123 \, [V]$$


##
$$u_B(I_1) = 0.004858979865499892 \, [A]$$


#####

<u>Incertidumbre de $V_1$ e $I_1$:</u>

#####

Se obtienen las incertidumbres de $V_1$ e $I_1$ a través de la siguiente expresión:

#####

\begin{equation*}
u(X_i) = \sqrt{ u_A^2(X_i) + u_B^2(X_i) }
\end{equation*}

#####

In [84]:
#def incertidumbre_combinada (uA, uB):
uC_V1 = incertidumbre_combinada (uA_V1, uB_V1)
uC_I1 = incertidumbre_combinada (uA_I1, uB_I1)

display(Markdown("##\n" + f"$$u(V_1) = {uC_V1} \, [V]$$\n"))
display(Markdown("##\n" + f"$$u(I_1) = {uC_I1} \, [A]$$\n"))

##
$$u(V_1) = 0.006489903543196924 \, [V]$$


##
$$u(I_1) = 0.00496685869874847 \, [A]$$


#####

Cálculo de la incertidumbre combinada:

#####

\begin{equation*} 
u_C(\bar{R_1}) = \sqrt{\left[\frac{dR_1}{dV_1}\right]^2 \cdot u^2(\bar{V_1}) + \left[\frac{dR_1}{dI_1}\right]^2 \cdot u^2(\bar{I_1}) + 2 \cdot \left[\frac{dR_1}{dV_1}\right] \cdot u(\bar{V_1}) \cdot \left[\frac{dR_1}{dI_1}\right] \cdot u(\bar{I_1}) \cdot r(\bar{V_1},\bar{I_1})}
\end{equation*}

#####

\begin{equation*} 
u_C(\bar{R_1}) = \sqrt{\left[\frac{1}{\bar{I_1}}\right]^2 \cdot u^2(\bar{V_1}) + \left[-\frac{\bar{V_1}}{\bar{I_1}^2}\right]^2 \cdot u^2(\bar{I_1}) + 2 \cdot \left[\frac{1}{\bar{I_1}}\right] \cdot u(\bar{V_1}) \cdot \left[-\frac{\bar{V_1}}{\bar{I_1}^2}\right] \cdot u(\bar{I_1}) \cdot r(\bar{V_1},\bar{I_1})}
\end{equation*}

#####

In [86]:
r_V1_I1 = 0.63

dR1_dV1 = 1/ov_I1
dR1_dI1 = -ov_V1/(ov_I1)**2

uC_R1 = np.sqrt( dR1_dV1**2 * uC_V1**2 + 
                 dR1_dI1**2 * uC_I1**2 +
                 2 * dR1_dV1 * uC_V1 * dR1_dI1 * uC_I1 * r_V1_I1)

display(Markdown("##\n" + f"$$u_C(R_1) = {uC_R1} \, [\\Omega]$$\n"))

##
$$u_C(R_1) = 0.028270849686331628 \, [\Omega]$$


#####

<u>Incertidumbre expandida:</u>

#####

\begin{equation*}
U(R_1) = u_C(R_1) \cdot k_{95\%}
\end{equation*}

#####

In [89]:
k_95 = 2
U_R1 = uC_R1 * k_95

display(Markdown("##\n" + f"$$U(R_1) = {U_R1} \, [\\Omega]$$\n" + "##"))

##
$$U(R_1) = 0.056541699372663255 \, [\Omega]$$
##

#####

<u>Resultado de la medición:</u>

#####

\begin{equation*}
R_1 = \overline{R_1} \pm U(R_1) 
\end{equation*}

#####

\begin{equation*}
R_1 = \left( 3.734121122599704 \pm 0.056541699372663255 \right) [\Omega] \qquad k = 2
\end{equation*}

#####

\begin{equation*}
R_1 = \left( 3.734 \pm 0.057 \right) [\Omega] \qquad k = 2
\end{equation*}

#####

***

<img src="Ejercicio_5.png" style="margin-left:auto; margin-right:auto"/>

***
#####

#### <u>Cálculo de valor más probable</u>

#####

Se obtienen los valores más probables de $V_1$ y $I_1$ a través de la siguiente expresión:

#####

\begin{equation*}
\overline{X_i} = \frac{1}{n} \cdot \sum\limits_{k=1}^{n} X_{i,k}
\end{equation*}

#####

Siendo:

> n: cantidad de mediciones

> $X_{i_{k}}$: i-ésimos valores de tensiones y/o corrientes, k-ésimos valores de las n mediciones.

#####

Reemplazando los valores de la tabla, obtenemos $\overline{R_1}$:

#####

In [176]:
H0 = [0.145, 0.123, 0.145, 0.145, 0.143]     #[m]
H1 = [4.525, 4.524, 4.521, 4.553, 4.551]     #[m]
RG = [10.1, 11.1, 11.2, 12.1, 11.2]          #[\Omega]

ov_H0 = valor_mas_probable(H0)
ov_H1 = valor_mas_probable(H1)
ov_RG = valor_mas_probable(RG)

display(Markdown("##\n" + f"$$\\overline{{H_0}} = {ov_H0} \, [m]$$\n"))
display(Markdown("##\n" + f"$$\\overline{{H_1}} = {ov_H1} \, [m]$$\n"))
display(Markdown("##\n" + f"$$\\overline{{R_G}} = {ov_RG} \, [\Omega]$$\n"))

##
$$\overline{H_0} = 0.14020000000000002 \, [m]$$


##
$$\overline{H_1} = 4.5348 \, [m]$$


##
$$\overline{R_G} = 11.14 \, [\Omega]$$


#####

Se obtiene el valor más probable de $V_0$ a través de la siguiente expresión:

#####

\begin{equation*}
\overline{V_0} = \left( \overline{H_1} - \overline{H_0} \right) \cdot K_H \cdot S \cdot \overline{A_V}
\end{equation*}

#####

\begin{equation*}
\overline{V_0} = \left( \overline{H_1} - \overline{H_0} \right) \cdot K_H \cdot S \cdot \left( 1 + \frac{ 50 k\Omega }{ \overline{R_G} } \right)
\end{equation*}

#####

Reemplazando los valores de la tabla, obtenemos $\overline{V_{0}}$:

#####


In [179]:
KH = 9.810                                   #[kPa/m]
S = 0.4                                      #[mV/kPa]

#def calculo_V0 (H1, H0, KH, S, RG)
ov_V0 = calculo_V0 (ov_H1, ov_H0, KH, S, ov_RG)

display(Markdown("##\n" + f"$$\\overline{{V_0}} = {ov_V0} \, [mV]$$\n"))

##
$$\overline{V_0} = 77415.85482332639 \, [mV]$$


#####

#### <u>Cálculo de incertidumbres</u>

#####

<u>Incertidumbre tipo A:</u>

#####

Se obtienen las incertidumbres tipo A a través de la siguiente expresión:

#####

\begin{equation*}
u_A(X_i) = \frac{ s(X_i) }{ \sqrt{n} } = \frac{ \sqrt{ \frac{ 1 }{ n-1 } \sum \limits_{k=1}^{n} \left( X_{i,k} - \overline{X_i} \right)^{2} } }{ \sqrt{n} }
\end{equation*}

#####

In [116]:
uA_H0 = incertidumbre_tipo_A (H0)
uA_H1 = incertidumbre_tipo_A (H1)
uA_S = 0
uA_RG = incertidumbre_tipo_A (RG)

display(Markdown("##\n" + f"$$u_A(H_0) = {uA_H0} \, [m]$$\n"))
display(Markdown("##\n" + f"$$u_A(H_1) = {uA_H1} \, [m]$$\n"))
display(Markdown("##\n" + f"$$u_A(S) = {uA_S} \, \\left[\\frac{{mV}}{{kPa}}\\right]$$\n"))
display(Markdown("##\n" + f"$$u_A(R_G) = {uA_RG} \, [\\Omega]$$\n"))

##
$$u_A(H_0) = 0.004317406628984579 \, [m]$$


##
$$u_A(H_1) = 0.007059745037889109 \, [m]$$


##
$$u_A(S) = 0 \, \left[\frac{mV}{kPa}\right]$$


##
$$u_A(R_G) = 0.3171750305430741 \, [\Omega]$$


#####

<u>Incertidumbre tipo B:</u>

#####

Se obtienen las incertidumbres tipo B a través de la siguiente expresión:

#####

\begin{equation*}
u_B(X_i) = \frac{ \Delta X_i }{ \sqrt{3} } = \frac{ \left( \frac{er\%}{100} + \frac{n}{ci} \right) \cdot \overline{X_i} }{ \sqrt{3} } 
\end{equation*}

#####


Siendo:

> $er%$: Error relativo.

> $n$: Error de cuentas.

> $ci$: Cuentas indicadas.

#####

Además:

> $ci = \frac{Ind}{Res} \frac{ [u] }{ \left[ \frac{u}{cuenta} \right] }= [cuentas]$

> $Ind$: Indicación del instrumento con su respectiva $u$ unidad.

> $Res$: Resolución del instrumento con su respectiva $u$ unidad por cuenta, dada por el rango seleccionado y la cantidad de dígitos que el mismo puede representar.

#####

In [117]:
er_H0 = 0.3
n_H0 = 0

ind_H0 = ov_H0
res_H0 = 0.001
ci_H0 = ind_H0 / res_H0

er_H1 = 0.3
n_H1 = 0

ind_H1 = ov_H1
res_H1 = 0.001
ci_H1 = ind_H1 / res_H1

er_S = 3.75
n_S = 0

ind_S = ov_S
res_S = 0.1
ci_S = ind_S / res_S

er_RG = 0.5
n_RG = 2

ind_RG = ov_RG
res_RG = 0.1
ci_RG = ind_RG / res_RG

#def incertidumbre_tipo_B (error_relativo, error_cuentas, cuentas_indicadas, valor_mas_probable):
uB_H0 = incertidumbre_tipo_B (er_H0, n_H0, ci_H0, ov_H0)
uB_H1 = incertidumbre_tipo_B (er_H1, n_H1, ci_H1, ov_H1)
uB_S = incertidumbre_tipo_B (er_S, n_S, ci_S, ov_S)
uB_RG = incertidumbre_tipo_B (er_RG, n_RG, ci_RG, ov_RG)

display(Markdown("##\n" + f"$$u_B(H_0) = {uB_H0} \, [m]$$\n"))
display(Markdown("##\n" + f"$$u_B(H_1) = {uB_H1} \, [m]$$\n"))
display(Markdown("##\n" + f"$$u_B(S) = {uB_S} \, \\left[\\frac{{mV}}{{kPa}}\\right]$$\n"))
display(Markdown("##\n" + f"$$u_B(R_G) = {uB_RG} \, [\\Omega]$$\n"))

##
$$u_B(H_0) = 0.00024283352322115667 \, [m]$$


##
$$u_B(H_1) = 0.007854504002163344 \, [m]$$


##
$$u_B(S) = 0.008660254037844387 \, \left[\frac{mV}{kPa}\right]$$


##
$$u_B(R_G) = 0.14762846383178735 \, [\Omega]$$


#####

<u>Incertidumbre de $V_1$ e $I_1$:</u>

#####

Se obtienen las incertidumbres de $V_1$ e $I_1$ a través de la siguiente expresión:

#####

\begin{equation*}
u(X_i) = \sqrt{ u_A^2(X_i) + u_B^2(X_i) }
\end{equation*}

#####

In [118]:
#def incertidumbre_combinada (uA, uB):
uC_H0 = incertidumbre_combinada (uA_H0, uB_H0)
uC_H1 = incertidumbre_combinada (uA_H1, uB_H1)
uC_S = incertidumbre_combinada (uA_S, uB_S)
uC_RG = incertidumbre_combinada (uA_RG, uB_RG)

display(Markdown("##\n" + f"$$u(H_0) = {uC_H0} \, [m]$$\n"))
display(Markdown("##\n" + f"$$u(H_1) = {uC_H1} \, [m]$$\n"))
display(Markdown("##\n" + f"$$u(S) = {uC_S} \, \\left[\\frac{{mV}}{{kPa}}\\right]$$\n"))
display(Markdown("##\n" + f"$$u(R_G) = {uC_RG} \, [\\Omega]$$\n"))

##
$$u(H_0) = 0.004324230350016057 \, [m]$$


##
$$u(H_1) = 0.010560929557572093 \, [m]$$


##
$$u(S) = 0.008660254037844387 \, \left[\frac{mV}{kPa}\right]$$


##
$$u(R_G) = 0.3498487720906468 \, [\Omega]$$


#####

<u> Incertidumbre combinada </u>

#####

Se obtiene la incertidumbre combinada de $V_0$ a través de la siguiente expresión:

#####

\begin{equation*} 
u_C(V_0) = \sqrt{\left[\frac{dV_0}{dH_0}\right]^2 \cdot u^2(H_0) + \left[\frac{dV_0}{dH_1}\right]^2 \cdot u^2(H_1) + \left[\frac{dV_0}{dS}\right]^2 \cdot u^2(S) + \left[\frac{dV_0}{dR_G}\right]^2 \cdot u^2(R_G) }
\end{equation*}

#####

\begin{equation*}
u_C(V_0) = \sqrt{ \left[ -K_H \cdot S \cdot A_V \right]^2 \cdot u^2(H_0) + \left[ K_H \cdot S \cdot A_V \right]^2 \cdot u^2(H_1) + \left[ \left( H_1 - H_0 \right) \cdot K_H \cdot A_V \right]^2 \cdot u^2(S) + \left[ \frac{ \left( H_1 - H_0 \right) \cdot K_H \cdot S \cdot 50 k\Omega }{ {R_G}^2 } \right]^2 \cdot u^2(R_G) }
\end{equation*}

#####

In [119]:
ov_AV = 1 + 50*10**3 / ov_RG

dV0_dH0 = -KH * S * ov_AV
dV0_dH1 = KH * S * ov_AV
dV0_dS = (ov_H1 - ov_H0) * KH * ov_AV
dV0_dRG = (ov_H1 - ov_H0) * KH * S * 50*10**3 / ov_RG**2

uC_V0 = np.sqrt( dV0_dH0**2 * uC_H0**2 + 
                 dV0_dH1**2 * uC_H1**2 +
                 dV0_dS**2 * uC_S**2 +
                 dV0_dRG**2 * uC_RG**2 )

display(Markdown("##\n" + f"$$u_C(V_0) = {uC_V0} \, [mV]$$\n"))

##
$$u_C(V_0) = 2959.3840538920354 \, [mV]$$


#####

<u>Incertidumbre expandida:</u>

#####

\begin{equation*}
U(V_0) = u_C(V_0) \cdot k_{95\%}
\end{equation*}

#####

In [122]:
k_95 = 2
U_V0 = uC_V0 * k_95

display(Markdown("##\n" + f"$$U(V_0) = {U_V0} \, [mV]$$\n" + "##"))

##
$$U(V_0) = 5918.768107784071 \, [mV]$$
##

#####

<u>Resultado de la medición:</u>

#####

\begin{equation*}
V_0 = \overline{V_0} \pm U(V_0) 
\end{equation*}

#####

\begin{equation*}
V_0 = \left( 77415.85482332639 \pm 5918.768107784071 \right) [mV] \qquad k = 2
\end{equation*}

#####

\begin{equation*}
\boxed{ V_0 = \left( 77.4 \pm 5.9 \right) [V] \qquad k = 2 }
\end{equation*}

#####

***

In [213]:
datos = {
    "Ejercicio": [1, 2, 3, 4, 5],
    "Resultado": [[ov_em, U_em], [ov_DeltaV, U_DeltaV], [ov_Vout, U_Vout], [ov_R1, U_R1], [round(ov_V0/1000, 1), round(U_V0/1000, 1)]],
    "k": [ k_95, k_95, k_95, k_95, k_95 ],
    "Tipo": ["Aproximación general", "Aproximación general", "Aproximación general", "Aproximación general", "Aproximación general"]
}

df = pd.DataFrame(datos)

formatos_por_ejercicio = {
    1: r"$ \left( \, {0[0]:.3f} \pm {0[1]:.3f} \, \right) $",
    2: r"$ \left( \, {0[0]:.2f} \pm {0[1]:.2f} \, \right) \, V$",
    3: r"$ \left( \, {0[0]:.2f} \pm {0[1]:.2f} \, \right) \, V$",
    4: r"$ \left( \, {0[0]:.3f} \pm {0[1]:.3f} \, \right) \, \Omega$",
    5: r"$ \left( \, {0[0]:.1f} \pm {0[1]:.1f} \, \right) \, V$",
}

df["Resultado"] = df.apply(
    lambda fila: formatos_por_ejercicio[fila["Ejercicio"]].format(fila["Resultado"]),
    axis=1
)

display(df.style.hide(axis="index"))